generic.go

```diff
@@ -62,12 +66,45 @@ func NewGeneric(config *Config, logger *zap.Logger, queue workqueue.RateLimiting
 			ZapNameField: descr.ZapNameField,
 		})
 		controllers[descr.Gvk] = iface
+
+		// Extra controller data
+		objectName := strings.Replace(
+			descr.Gvk.Group+"_"+descr.Gvk.Version+"_"+descr.Gvk.Kind,
```

Comment:
This is for the metric name - I'm concerned it might be possible for the implementation to pass controllers that handle different two versions of the same GroupKind. The alternative can be to share the metrics (create a holder for GroupKind instead of Gvk)

Reply:
Versions are just a way to represent the same backing object. Controllers do not need to work with multiple versions of objects, only one is enough. E.g. if you send a create with v1 then you can read with v2. And vice versa.